src/promptflow-core/promptflow/contracts/flow.py

```diff
@@ -1019,6 +1026,46 @@ def deserialize(data: dict) -> "FlexFlow":
             environment_variables=data.get("environment_variables") or {},
         )
 
+    @classmethod
+    def _from_dict(cls, flow_dag: dict, working_dir: Path, name=None) -> "FlexFlow":
+        """Load flow from dict."""
+        from promptflow._core.entry_meta_generator import generate_flow_meta
+
+        from .._utils.flow_utils import resolve_python_entry_file
+
+        Flow._update_working_dir(working_dir)
+        if name is None:
+            name = flow_dag.get("name", _sanitize_python_variable_name(working_dir.stem))
+        flow_dag["name"] = name
+
+        entry = flow_dag.get("entry")
+        entry_file = resolve_python_entry_file(entry=entry, working_dir=working_dir)
+
+        meta_dict = generate_flow_meta(
+            flow_directory=working_dir,
+            source_path=entry_file,
+            data=flow_dag,
+        )
+        return cls.deserialize(meta_dict)
+
+    def get_connection_names(self, environment_variables_overrides: Dict[str, str] = None):
+        """Return connection names."""
+        from promptflow.core._model_configuration import MODEL_CONFIG_NAME_2_CLASS
+
+        connection_names = super().get_connection_names(environment_variables_overrides=environment_variables_overrides)
+        try:
```

Comment:
Do we still need to get connection names from `PROMPTFLOW_FLOW_INIT_CONFIG`?

Reply:
Yes, this will make the code logic more complete, and if the environment variable is set during build, connections can also be generated:
![image](https://github.com/microsoft/promptflow/assets/23182548/8e2d7bb8-e885-46af-89b4-379af7ab3ff7)

Reply:
Changing this will impact other scenarios (flow test/ flow run). I'm not sure if a connection not found error should be raised when connection in `PROMPTFLOW_FLOW_INIT_CONFIG` not exist.

Reply:
I have deleted these codes.